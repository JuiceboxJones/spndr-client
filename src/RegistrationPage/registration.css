.registration_container{
  width: 33%;
  margin: 0 auto;
  margin-top: 10%;
  border-radius: 8px;
  background: #F5F5F5;
}

.registration_container h2{
  background: #E0DFDF;
  height: 32px;
  margin: 0;
  padding: 10px;
  border-radius: 8px 8px 0 0;
}

.registration_container label{
  display: inline-block;
  text-align: right;
  margin: 10px;
  
}

.registration_container input{
  margin: 5px;
}

.registrationForm{
  width: 50%;
  margin: 0 auto;
}
#register_button{
  margin: 10px 0 20px 0;
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background:#A7C4C0;
  cursor: pointer;
  font-weight: bold;
}
#register_button:hover{
  background: #024e2e9c;
  color:white;
}
.registration_input_container{
  width: 35%;
  margin: 0 auto;
}

@media screen and (max-width: 480px){
  .registration_container{
    width: 90%;
  }

  .registration_input_container{
    width: 63%;
  }
  
}
