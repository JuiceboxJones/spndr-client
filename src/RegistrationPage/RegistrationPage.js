import React, { Component } from 'react';
import RegistrationForm from '../RegistrationForm/RegistrationForm';
import Header from '../Header/Header';
import './registration.css';

export default class RegistrationPage extends Component {
  static defaultProps = {
    history: {
      push: () => {}
    }
  };

  handleRegistrationSuccess = user => {
    const { history } = this.props;
    history.push('/welcome');
  };

  render() {
    return (
      <div>
        <Header />
        <div className="registration_container">
          <section className="RegistrationPage">
            <h2>Register</h2>
            <RegistrationForm
              onRegistrationSuccess={this.handleRegistrationSuccess}
            />
          </section>
        </div>
      </div>
    );
  }
}
