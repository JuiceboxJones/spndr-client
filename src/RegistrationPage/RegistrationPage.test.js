import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationPage from './RegistrationPage'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RegistrationPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<RegistrationPage />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});