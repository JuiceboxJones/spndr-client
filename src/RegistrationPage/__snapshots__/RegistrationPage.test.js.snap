// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="header"
  >
    <header>
      <h1>
        Welcome to SPNDR
      </h1>
    </header>
    <p>
      Finance Management. Simplified.
    </p>
  </div>
  <div
    className="registration_container"
  >
    <section
      className="RegistrationPage"
    >
      <h2>
        Register
      </h2>
      <form
        className="RegistrationForm"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="full_name"
        >
          <label
            htmlFor="RegistrationForm__full_name"
          >
            Full name
          </label>
          <input
            id="RegistrationForm__full_name"
            name="full_name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="user_name"
        >
          <label
            htmlFor="RegistrationForm__user_name"
          >
            User name
          </label>
          <input
            id="RegistrationForm__user_name"
            name="user_name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="password"
        >
          <label
            htmlFor="RegistrationForm__password"
          >
            Password
          </label>
          <input
            id="RegistrationForm__password"
            name="password"
            required={true}
            type="password"
          />
        </div>
        <button
          type="submit"
        >
          Register
        </button>
      </form>
    </section>
  </div>
</div>
`;
